### PR TITLE
Refactoring restore tools so we can inject any output stream.

### PR DESCRIPTION
**Pull request by mneedham**
This is specifically so that we can choose to capture System.out or not
in testing.

### CODE REVIEW

File: enterprise/core-edge/src/main/java/org/neo4j/restore/RestoreExistingClusterCli.java [hunk 1]
```diff
@@ -46,12 +46,29 @@
 
 public class RestoreExistingClusterCli
 {
+    private final PrintStream out;
+
+    public RestoreExistingClusterCli()
+    {
+        this(System.out);
```
Comment (davidegrohmann) [inline enterprise/core-edge/src/main/java/org/neo4j/restore/RestoreExistingClusterCli.java]: code style

File: enterprise/core-edge/src/main/java/org/neo4j/restore/RestoreExistingClusterCli.java [hunk 2]
```diff
@@ -46,12 +46,29 @@
...
+    {
+        this.out = out;
+    }
+
     public static void main( String[] incomingArguments )
     {
+        new RestoreExistingClusterCli(  ).run( incomingArguments );
+    }
+
+    public void run(String[] incomingArguments)
```
Comment (davidegrohmann) [inline enterprise/core-edge/src/main/java/org/neo4j/restore/RestoreExistingClusterCli.java]: code style

File: enterprise/core-edge/src/main/java/org/neo4j/restore/RestoreNewClusterCli.java [hunk 3]
```diff
@@ -47,12 +47,29 @@
...
+    private final PrintStream out;
+
+    public RestoreNewClusterCli( PrintStream out )
+    {
+        this.out = out;
+    }
+
+    public RestoreNewClusterCli( )
+    {
+        this(System.out);
```
Comment (davidegrohmann) [inline enterprise/core-edge/src/main/java/org/neo4j/restore/RestoreNewClusterCli.java]: code style

**Comment by mneedham** [comment]
@davidegrohmann  fixed. WIll merge

**Pull request merged**